src/lunchselector/model.clj

```diff
@@ -39,25 +42,49 @@
 (defn- get-rest-rating [restaurant]
   (:aggregate_rating (:user_rating restaurant)))
 
-(defn get-restaurants [keyword]
+(defn online-restaurants [keyword]
   (loop [restaurant-list (:restaurants
-                          (cheshire/parse-string
-                           (:body (fetch-restaurants keyword)) true))
+                          (lu/parse-to-clj-map
+                           (fetch-restaurants keyword)))
          resta ()]
     (if (empty? restaurant-list)
       resta
       (recur (rest restaurant-list)
              (conj resta (get-rest-name (first restaurant-list)))))))
 
-(defn submit-votes [user votes]
+(defn- cast-new-vote? [email restaurant]
+  (let [user-id (:user_id (first (ldb/fetch-user-id email)))
+        rest-id (:rest_id (first (ldb/fetch-restaurant-id restaurant)))]
+    (if (empty? (ldb/check-votes user-id rest-id))
+      true
+      false)))
+
+(defn submit-votes [email votes]
   (if (vector? votes)
     (doseq [x votes]
-      (try
-        (ldb/insert {:user user :restaurant x})
-        (catch Exception e
-          (prn e))))
-
-    (try
-      (ldb/insert {:user user :restaurant votes})
-        (catch Exception e
-          (prn e)))))
+      (ldb/cast-vote-safe {:email email :restaurant x}))
+    (ldb/cast-vote-safe {:email email :restaurant votes})))
+
+(defn submit-users [user email]
+  (ldb/add-user-safe {:name user :email email}))
+
+(defn submit-restaurants [rest added-by]
+  (if (vector? rest)
```

Comment:
What is `rest`?  Also, you probably want `coll?` or `sequential?`; it's not preferable to bind implementation to a type of sequence.